version: '3.8'

services:
  api:
    build: ./VetManagerAPI
    command: python manage.py runserver 0.0.0.0:8000
    volumes:
      - ./VetManagerAPI:/app
    ports:
      - "8000:8000"
    depends_on:
      db:
        condition: service_healthy
    environment:
      - DEBUG=${DEBUG}
      - DB_NAME=${DB_NAME}
      - DB_USER=${DB_USER}
      - DB_PASSWORD=${DB_PASSWORD}
      - DB_HOST=${DB_HOST}
      - DB_PORT=${DB_PORT}
    networks:
      - vetmanager_network

  db:
    image: ${DB_IMAGE}
    volumes:
      - ${DB_VOLUME_PATH}
    environment:
      POSTGRES_DB: ${DB_NAME}
      POSTGRES_USER: ${DB_USER}
      POSTGRES_PASSWORD: ${DB_PASSWORD}
    ports:
      - "${DB_PORT}:${DB_PORT}"
    networks:
      - vetmanager_network
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${DB_USER} -d ${DB_NAME}"]
      interval: 10s
      retries: 5

volumes:
  postgres_data:

networks:
  vetmanager_network:
